### ISSUE TITLE
90% of bins only contain a single contig

### ISSUE THREAD

**Issue by liangjinsong** [RasmussenLab/vamb#124]
Hi, I used snakemake to cluster 16686 contigs (>2kb) of 400 soil sample. The log file shows that "Clustered 16686 contigs in 15068 bins", it means that 90% of bins only contain a single contig. Could you please help find the reason?

---------snakemake conmand----------
snakemake --cores 192 --configfile /NFSfile/combined_contigs/config.json --snakefile /home/jinsong/software/vamb-3.0.7/workflow/vamb.snake.conda.py
------------------------------------------

---------config.json----------
{
   "contigs": "/NFSfile/combined_contigs/contigs.txt",
   "sample_data": "/NFSfile/fastq_file/samples2data.txt",
   "index_size": "2500G",
   "minimap_mem": "1500gb",
   "minimap_ppn": "190",
   "vamb_mem": "150gb",
   "vamb_ppn": "190",
   "checkm_mem": "2500gb",
   "checkm_ppn": "190",
   "vamb_params": "-o C -m 2000 --minfasta 2000 --outdir vamb"
}
------------------------------------------

---------contigs.txt----------
/NFSfile/combined_contigs/contigs_ALM105.fa
/NFSfile/combined_contigs/contigs_ALM154.fa
/NFSfile/combined_contigs/contigs_CM429.fa
/NFSfile/combined_contigs/contigs_ALM145.fa
******************************************************
------------------------------------------

---------samples2data.txt----------
ALM105  /NFSfile/fastq_file/ALM105_1.fastq.gz        /NFSfile/fastq_file/ALM105_2.fastq.gz
ALM154  /NFSfile/fastq_file/ALM154_1.fastq.gz        /NFSfile/fastq_file/ALM154_2.fastq.gz
CM429   /NFSfile/fastq_file/CM429_1.fastq.gz /NFSfile/fastq_file/CM429_2.fastq.gz
ALM145.fa       /NFSfile/fastq_file/ALM145_1.fastq.gz        /NFSfile/fastq_file/ALM145_2.fastq.gz
******************************************************
------------------------------------------

---------log.txt ----------
Starting Vamb version 3.0.3
        Date and time is 2022-05-07 15:38:49.760185

Loading TNF
        Minimum sequence length: 2000
        Loading data from FASTA file contigs.flt.fna.gz

        Kept 49100442 bases in 16686 sequences
        Processed TNF in 14.72 seconds

Loading RPKM
        Reference hash: 156282df3f4cb0cfe218fb44e6e74965
        Loading RPKM from JGI file jgi_matrix/jgi.abundance.dat
        Processed RPKM in 0.51 seconds

Creating and training VAE
        Created VAE
        Created dataloader and mask
        Number of sequences unsuitable for encoding: 0
        Number of sequences remaining: 16686

        Network properties:
        CUDA: False
        Alpha: 0.15
        Beta: 200.0
        Dropout: 0.2
        N hidden: 512, 512
        N latent: 32

        Training properties:
        N epochs: 500
        Starting batch size: 256
        Batchsteps: 25, 75, 150, 300
        Learning rate: 0.001
        N sequences: 16686
        N samples: 3

        Epoch: 1        Loss: 0.465505  CE: 0.4541577   SSE: 70.846592  KLD: 70.0609    Batchsize: 256
        Epoch: 2        Loss: 0.282252  CE: 0.2533835   SSE: 47.876919  KLD: 105.4993   Batchsize: 256
        Epoch: 3        Loss: 0.267680  CE: 0.2456349   SSE: 42.812842  KLD: 97.8067    Batchsize: 256
        Epoch: 4        Loss: 0.259438  CE: 0.2415106   SSE: 40.175405  KLD: 90.0614    Batchsize: 256
        Epoch: 5        Loss: 0.254942  CE: 0.2407871   SSE: 38.194738  KLD: 83.3310    Batchsize: 256
        ***********************************************************************************************
        Epoch: 496      Loss: 0.211842  CE: 0.2125962   SSE: 25.496797  KLD: 65.4361    Batchsize: 4096
        Epoch: 497      Loss: 0.212047  CE: 0.2129362   SSE: 25.446844  KLD: 65.5325    Batchsize: 4096
        Epoch: 498      Loss: 0.211847  CE: 0.2126688   SSE: 25.502120  KLD: 65.0603    Batchsize: 4096
        Epoch: 499      Loss: 0.212146  CE: 0.2131529   SSE: 25.497944  KLD: 64.6161    Batchsize: 4096
        Epoch: 500      Loss: 0.211828  CE: 0.2128435   SSE: 25.435772  KLD: 64.6922    Batchsize: 4096

        Encoding to latent representation
        Trained VAE and encoded in 12073.63 seconds

Clustering
        Windowsize: 200
        Min successful thresholds detected: 20
        Max clusters: None
        Min cluster size: 1
        Use CUDA for clustering: False
        Separator: "C"

        Clustered 16686 contigs in 15068 bins
        Clustered contigs in 11.24 seconds

Writing FASTA files
        Minimum FASTA size: 2000

        Wrote 16686 contigs to 15068 FASTA files
        Wrote FASTA in 2.74 seconds

Completed Vamb in 12102.95 seconds
------------------------------------------

**Comment by jakobnissen** [comment]
If assembling 400 samples only leads to 16k contigs, then I'd guess that either the data is too complex to assemble correctly, so there are simply no complete genomes in your data and the few contigs you have are indeed from different organisms.
My advice would be to somehow improve the assembly - having less than 5 Mbp assembly from 400 samples is just not good enough to bin.
Closing as solved - you're welcome to re-open if you have another question
